tests/test_analysis.py

```diff
@@ -318,6 +318,47 @@ def test_calc_sem(self):
         SEM = aopy.analysis.calc_sem(data, axis=(0,2))
         np.testing.assert_allclose(SEM, np.nanstd(data, axis=(0,2))/np.sqrt(18) )
 
+    def test_calc_mean_erp(self):
+        nevents = 3
+        event_times = 0.2 + np.arange(nevents)
+        samplerate = 1000
+        nch = 2
+        data = np.zeros(((1+nevents)*samplerate, nch))
+
+        # Make the data zero everywhere except for one sample after each event time
+        print([int(t)+1 for t in event_times*samplerate])
+        data[[int(t)+1 for t in event_times*samplerate],0] = 1
+        data[[int(t)+1 for t in event_times*samplerate],1] = 2
+
+        self.assertEqual(np.sum(data[:,0]), nevents)
+        self.assertEqual(np.sum(data[:,1]), nevents*2)
+
+        mean_erp = aopy.analysis.calc_mean_erp(data, event_times, 0.1, 0.1, samplerate, subtract_baseline=False)
+        self.assertEqual(np.sum(mean_erp[:,0]), 1)
+        self.assertEqual(np.sum(mean_erp[:,1]), 2)
+
+        # Subtract baseline
+        data += 1
+        mean_erp = aopy.analysis.calc_mean_erp(data, event_times, 0.1, 0.1, samplerate)
+        self.assertEqual(np.sum(mean_erp[:,0]), 1)
+        self.assertEqual(np.sum(mean_erp[:,1]), 2)
+
+    def test_calc_max_erp(self):
+        nevents = 3
+        event_times = 0.2 + np.arange(nevents)
+        samplerate = 1000
+        nch = 2
+        data = np.zeros(((1+nevents)*samplerate, nch))
+
+        # Make the data zero everywhere except for one sample after each event time
+        data[[int(t)+1 for t in event_times*samplerate],0] = 1
+        data[[int(t)+1 for t in event_times*samplerate],1] = 2
+
+        max_erp = aopy.analysis.calc_max_erp(data, event_times, 0.1, 0.1, samplerate)
+        self.assertTrue(max_erp[0] > 0.5) 
+        self.assertTrue(max_erp[1] > 1.)
+        # not sure why the max erp isn't exactly 1 (or 2 for ch2), it seems to get spread out over a couple samples
```

Comment:
that seems concerning. Could you show some plots of the test data (raw + trial-aligned) so we can go over this? Seems like something to sort out.

Reply:
fixed the rounding error that was causing this. probably not a big deal at 1khz or 25khz sampled data to be off by one sample, but anyway good that it works as expected now.